File: app/src/main/java/org/techtown/find_gas_station/set/SetDao.java
package org.techtown.find_gas_station.set;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface SetDao {

    @Query("SELECT * FROM set_table")
    Set getAll();

    @Query("SELECT * FROM set_table")
    LiveData<Set> getAllLiveData();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Set set);

    @Query("DELETE FROM set_table")
    void deleteAll();
    //모든 데이터를 지운다.

    @Update
    void update(Set set);


}

File: app/src/main/java/org/techtown/find_gas_station/set/SetDao.kt
package org.techtown.find_gas_station.set

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update


@Dao
interface SetDao {

    @Query("SELECT * FROM set_table")
    fun getOilLocalData() : LiveData<Set>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(set : Set)

    @Query("DELETE FROM set_table")
    suspend fun deleteAll()

    @Update
    suspend fun update(set : Set)


}